File: backend-2024-grupo2/src/main/kotlin/ar/edu/unsam/phm/backendgrupo2/domain/Role.kt
package ar.edu.unsam.phm.backendgrupo2.domain

import ar.edu.unsam.phm.backendgrupo2.dto.RoleDTO
import jakarta.persistence.*

@Entity
@Table(name = "role")
class Role {

  companion object {

    fun fromJSON(roleDTO: RoleDTO): Role =
      Role().apply {
        this.id = roleDTO.id
        this.name = roleDTO.name
        this.isAdministrator = roleDTO.isAdministrator
      }
  }

  @Id
  @GeneratedValue
  var id: Int = 0

  @Column(length = 40)
  var name: String = ""

  @Column
  var isAdministrator: Boolean = false
}

File: frontend-2024-grupo2/src/app/data/models/Role.ts
export type RoleJSON = {
  name: string
  isAdministrator: boolean
}

export class Role {
  constructor(
    public name: string,
    public isAdministrator: boolean
  ) {}

  static fromJSON(roleJSON: RoleJSON): Role {
    return Object.assign(new Role(roleJSON.name, roleJSON.isAdministrator))
  }

  toJSON(): RoleJSON {
    return {
      name: this.name,
      isAdministrator: this.isAdministrator
    }
  }
}
